machine_types.go

```diff
@@ -581,7 +585,7 @@ type HTTPResponseOptions struct {
 type MachineService struct {
 	Protocol                 string                     `json:"protocol,omitempty" toml:"protocol,omitempty"`
 	InternalPort             int                        `json:"internal_port,omitempty" toml:"internal_port,omitempty"`
-	Autostop                 *bool                      `json:"autostop,omitempty"`
+	Autostop                 *MachineAutostop           `json:"autostop,omitempty"`
```

Comment:
I imagine you already considered and discussed the option to add a `AutostopMode` string field here.